code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,413 @@
+
+	var/on_store_message = "has entered long-term storage."
+	var/on_store_name = "Cryogenic Oversight"
+
+	// 3 minutes-ish safe period before being despawned.
+	var/time_till_despawn = 3 MINUTES // This is reduced to 30 seconds if a player manually enters cryo
+	var/despawn_world_time = null          // Used to keep track of the safe period.
+
+	var/obj/machinery/computer/cryopod/control_computer
+	COOLDOWN_DECLARE(last_no_computer_message)
+
+	// These items are preserved when the process() despawn proc occurs.
+	var/static/list/preserve_items = list(
+		/obj/item/hand_tele,
+		/obj/item/card/id/advanced/gold/captains_spare,
+		/obj/item/aicard,
+		/obj/item/mmi,
+		/obj/item/paicard,
+		/obj/item/gun,
+		/obj/item/pinpointer,
+		/obj/item/clothing/shoes/magboots,
+		/obj/item/areaeditor/blueprints,
+		/obj/item/clothing/head/helmet/space,
+		/obj/item/clothing/suit/space,
+		/obj/item/clothing/suit/armor,
+		/obj/item/defibrillator/compact,
+		/obj/item/reagent_containers/hypospray/cmo,
+		/obj/item/clothing/accessory/medal/gold/captain,
+		/obj/item/clothing/gloves/krav_maga,
+		/obj/item/nullrod,
+		/obj/item/tank/jetpack,
+		/obj/item/documents,
+		/obj/item/nuke_core_container
+	)
+	// These items will NOT be preserved
+	var/static/list/do_not_preserve_items = list (
+		/obj/item/mmi/posibrain
+	)
+
+/obj/machinery/cryopod/Initialize()
+	..()
+	return INITIALIZE_HINT_LATELOAD //Gotta populate the cryopod computer GLOB first
+
+/obj/machinery/cryopod/LateInitialize()
+	update_icon()
+	find_control_computer()
+
+// This is not a good situation
+/obj/machinery/cryopod/Destroy()
+	control_computer = null
+	return ..()
+
+/obj/machinery/cryopod/proc/find_control_computer(urgent = FALSE)
+
+	for(var/cryo_console as anything in GLOB.cryopod_computers)
+		var/obj/machinery/computer/cryopod/console = cryo_console
+		if(get_area(console) == get_area(src))
+			control_computer = console
+			break
+
+	// Don't send messages unless we *need* the computer, and less than five minutes have passed since last time we messaged
+	if(!control_computer && urgent && COOLDOWN_FINISHED(src, last_no_computer_message))
+		COOLDOWN_START(src, last_no_computer_message, 5 MINUTES)
+		log_admin("Cryopod in [get_area(src)] could not find control computer!")
+		message_admins("Cryopod in [get_area(src)] could not find control computer!")
+		last_no_computer_message = world.time
+
+	return control_computer != null
+
+/obj/machinery/cryopod/close_machine(mob/user)
+	if(!control_computer)
+		find_control_computer(TRUE)
+	if((isnull(user) || istype(user)) && state_open && !panel_open)
+		..(user)
+		var/mob/living/mob_occupant = occupant
+		if(mob_occupant && mob_occupant.stat != DEAD)
+			to_chat(occupant, "<span class='boldnotice'>You feel cool air surround you. You go numb as your senses turn inward.</span>")
+		if(mob_occupant.client || !mob_occupant.key) // Self cryos and SSD
+			despawn_world_time = world.time + (time_till_despawn * 0.1) // This gives them 30 seconds
```

Comment:
Doesn't this also mean if `time_till_despawn` is changed at all, then it's no longer 30 seconds? I think the 30 seconds should be its own variable/define.